### ISSUE TITLE
On adapting this method to a new dataset

### ISSUE THREAD

**Issue by BaophanN** [zhuoxiao-chen/MOS#4]
Dear author, thanks for this work. I am trying to adapt this method to a new custom dataset, but I still not understand the flow clearly. Please give me some guidance. 
Given that I have a dataset with training set as the source domain (has labels) and validation set (target domain) has no label. The source-only model was trained using ST3D model. How can I apply MOS method to it? What I need is to apply MOS to generate predictions on Validation set. As far as I understand, the flow of MOS is like this in terms of code: 
1. Pretrained the baseline on source domain for 6 epochs. 
2. Assemble the model into a super model using train.py, which also called **mos** through **tta_func** on target domain. In this case, what is the **PRETRAINED_MODEL**, is it the path to the checkpoint I want to evaluate. For example, /my-path/checkpoint_6.pth
```
python train.py --cfg_file cfgs/tta_w2k_models/secondiou/mos.yaml \
    --batch_size ${BATCH_SIZE} --pretrained_model ${PRETRAINED_MODEL}
```
4. Evaluate the checkpoint after TTA 
```
python test_tta.py --cfg_file cfgs/tta_w2k_models/secondiou/mos.yaml \
    --batch_size ${BATCH_SIZE} --eval_all
```
I have somewhat adapt your method to fit my current situation, however I have this problem. when I run step 2. This is the output from the terminal:

<img width="1894" height="70" alt="Image" src="https://github.com/user-attachments/assets/39349ec2-01c0-4a4b-946b-df2f764b389c" />

While when i run step 4, the get_no_evaluated_ckpt return -1, None
Here is the full config file of mine: 
```
CLASS_NAMES: ['Vehicle']

DATA_CONFIG:
    _BASE_CONFIG_: cfgs/dataset_configs/da_waymo_dataset.yaml
    PROCESSED_DATA_TAG: 'waymo_processed_data_v0_5_0'
    INFO_WITH_FAKELIDAR: False

    SAMPLED_INTERVAL: {
        'train': 5,
        'test': 10
    }

    DATA_AUGMENTOR:
        DISABLE_AUG_LIST: [ 'random_object_scaling', 'normalize_object_size' ]
        AUG_CONFIG_LIST:
            - NAME: gt_sampling
              USE_ROAD_PLANE: False
              DB_INFO_PATH:
                  - waymo_processed_data_v0_5_0_waymo_dbinfos_train_sampled_1.pkl
              PREPARE: {
                  filter_by_min_points: [ 'Vehicle:5' ],
                  filter_by_difficulty: [ -1 ],
              }

              SAMPLE_GROUPS: [ 'Vehicle:25' ]
              NUM_POINT_FEATURES: 5
              DATABASE_WITH_FAKELIDAR: True
              REMOVE_EXTRA_WIDTH: [ 0.0, 0.0, 0.0 ]
              LIMIT_WHOLE_SCENE: True

            - NAME: random_object_scaling
              SCALE_UNIFORM_NOISE: {
                  'Vehicle': [ 0.7, 0.95 ]
              }

            - NAME: normalize_object_size
              SIZE_RES: {
                  'Vehicle': [ -0.91, -0.49, -0.26 ]
              }

            - NAME: random_object_rotation
              ROT_PROB: 1.0
              ROT_UNIFORM_NOISE: [ -0.78539816, 0.78539816 ]

            - NAME: random_world_flip
              ALONG_AXIS_LIST: [ 'x', 'y' ]

            - NAME: random_world_rotation
              WORLD_ROT_ANGLE: [ -0.78539816, 0.78539816 ]

            - NAME: random_world_scaling
              WORLD_SCALE_RANGE: [ 0.95, 1.05 ]

DATA_CONFIG_TAR:
    _BASE_CONFIG_: cfgs/dataset_configs/da_kitti_dataset.yaml
    TARGET: True
    FOV_POINTS_ONLY: False
    CLASS_NAMES: ['Car']
    SHIFT_COOR: [0.0, 0.0, 1.6]
    TTA_DATA_AUGMENTOR:
        DISABLE_AUG_LIST: [ 'placeholder' ]
        AUG_CONFIG_LIST:
            - NAME: random_object_scaling
              SCALE_UNIFORM_NOISE: [ 0.7, 0.95 ]
            - NAME: random_world_scaling
              WORLD_SCALE_RANGE: [ 0.95, 1.05 ]

MODEL:
    NAME: SECONDNetIoU

    VFE:
        NAME: MeanVFE

    BACKBONE_3D:
        NAME: VoxelBackBone8x

    MAP_TO_BEV:
        NAME: HeightCompression
        NUM_BEV_FEATURES: 256

    BACKBONE_2D:
        NAME: BaseBEVBackbone

        LAYER_NUMS: [5, 5]
        LAYER_STRIDES: [1, 2]
        NUM_FILTERS: [128, 256]
        UPSAMPLE_STRIDES: [1, 2]
        NUM_UPSAMPLE_FILTERS: [256, 256]

    DENSE_HEAD:
        NAME: AnchorHeadSingle
        CLASS_AGNOSTIC: False

        USE_DIRECTION_CLASSIFIER: True
        DIR_OFFSET: 0.78539
        DIR_LIMIT_OFFSET: 0.0
        NUM_DIR_BINS: 2

        ANCHOR_GENERATOR_CONFIG: [
            {
                'class_name': 'Vehicle',
                'anchor_sizes': [[4.2, 2.0, 1.6]],
                'anchor_rotations': [0, 1.57],
                'anchor_bottom_heights': [0],
                'align_center': False,
                'feature_map_stride': 8,
                'matched_threshold': 0.55,
                'unmatched_threshold': 0.4
            }
        ]

        TARGET_ASSIGNER_CONFIG:
            NAME: AxisAlignedTargetAssigner
            POS_FRACTION: -1.0
            SAMPLE_SIZE: 512
            NORM_BY_NUM_EXAMPLES: False
            MATCH_HEIGHT: False
            BOX_CODER: ResidualCoder

        LOSS_CONFIG:
            LOSS_WEIGHTS: {
                'cls_weight': 1.0,
                'loc_weight': 2.0,
                'dir_weight': 0.2,
                'code_weights': [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
            }

    ROI_HEAD:
        NAME: SECONDHead
        CLASS_AGNOSTIC: True

        SHARED_FC: [256, 256]
        IOU_FC: [256, 256]
        DP_RATIO: 0.3

        NMS_CONFIG:
            TRAIN:
                NMS_TYPE: nms_gpu
                MULTI_CLASSES_NMS: False
                NMS_PRE_MAXSIZE: 9000
                NMS_POST_MAXSIZE: 512
                NMS_THRESH: 0.8
            TEST:
                NMS_TYPE: nms_gpu
                MULTI_CLASSES_NMS: False
                NMS_PRE_MAXSIZE: 1024
                NMS_POST_MAXSIZE: 128
                NMS_THRESH: 0.7

        ROI_GRID_POOL:
            GRID_SIZE: 7
            IN_CHANNEL: 512
            DOWNSAMPLE_RATIO: 8

        TARGET_CONFIG:
            BOX_CODER: ResidualCoder
            ROI_PER_IMAGE: 128
            FG_RATIO: 0.5

            SAMPLE_ROI_BY_EACH_CLASS: True
            CLS_SCORE_TYPE: raw_roi_iou

            CLS_FG_THRESH: 0.75
            CLS_BG_THRESH: 0.25
            CLS_BG_THRESH_LO: 0.1
            HARD_BG_RATIO: 0.8

            REG_FG_THRESH: 0.55

        LOSS_CONFIG:
            IOU_LOSS: BinaryCrossEntropy
            LOSS_WEIGHTS: {
                'rcnn_iou_weight': 1.0,
                'code_weights': [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
            }

    POST_PROCESSING:
        RECALL_THRESH_LIST: [0.3, 0.5, 0.7]
        SCORE_THRESH: 0.1
        OUTPUT_RAW_SCORE: False

        EVAL_METRIC: kitti

        NMS_CONFIG:
            MULTI_CLASSES_NMS: False
            NMS_TYPE: nms_gpu
            NMS_THRESH: 0.01
            NMS_PRE_MAXSIZE: 4096
            NMS_POST_MAXSIZE: 500

OPTIMIZATION:
    NUM_EPOCHS: 1
    OPTIMIZER: adam_onecycle
    LR: 0.001
    WEIGHT_DECAY: 0.01
    MOMENTUM: 0.9

    MOMS: [0.95, 0.85]
    PCT_START: 0.4
    DIV_FACTOR: 10
    DECAY_STEP_LIST: [35, 45]
    LR_DECAY: 0.1
    LR_CLIP: 0.0000001

    LR_WARMUP: False
    WARMUP_EPOCH: 1

    GRAD_NORM_CLIP: 10

TTA:
    ENABLE: TRUE
    METHOD: mos
    SAVE_CKPT: [0,2, 4,8, 16,32, 2100]
    SAVE_CKPT_INTERVAL: 112
    MOS_SETTING:
        AGGREGATE_START_CKPT: 32
        AGGREGATE_BANK: True
        BANK_START_CKPT: 192
        AGGREGATE_NUM: 5


SELF_TRAIN:
    SCORE_THRESH: [0.6]
    NEG_THRESH: [0.2]
    UPDATE_PSEUDO_LABEL: [0]
    UPDATE_PSEUDO_LABEL_INTERVAL: 2
    INIT_PS: None
    SRC:
        USE_DATA: False
        USE_GRAD: False
        LOSS_WEIGHT: 1.0
    TAR:
        USE_DATA: True
        LOSS_WEIGHT: 1.0

    PROG_AUG:
        ENABLED: False
        UPDATE_AUG: [0]
        SCALE: 1.2

    MEMORY_ENSEMBLE:
        ENABLED: False
        NAME: consistency_ensemble
        IOU_THRESH: 0.1

        NMS_CONFIG:
            NMS_TYPE: nms_gpu
            MULTI_CLASSES_NMS: False
            NMS_PRE_MAXSIZE: 512
            NMS_POST_MAXSIZE: 100
            NMS_THRESH: 0.1

        MEMORY_VOTING:
            ENABLED: True
            IGNORE_THRESH: 2
            RM_THRESH: 3


```
